validator/src/journal/block_validator.rs

```diff
@@ -363,6 +379,32 @@ where
                     self.validation_thread_exit.store(true, Ordering::Relaxed);
                 }
             }
+        } else {
+            for desc in pending_descendants_marked_for_processing {
+                match response_sender.send(BlockValidationResult {
+                    block_id: desc.header_signature.clone(),
+                    execution_results: vec![],
+                    num_transactions: 0,
+                    status: BlockStatus::Invalid,
+                }) {
+                    Ok(..) => {
+                        info!(
+                            "Block found invalid {}; marking descendant {} invalid",
+                            &block.header_signature[..8],
+                            &desc.header_signature[..8]
```

Comment:
On crb307-n1 which was deployed from a snapshot and was connected to testnet we see the following sequence of events in the logs:

```
[2022-01-28 15:28:43.594 DEBUG    completer] Request missing predecessor: 84a0180d9800b6556deddd91e79636f18b901b14611441b2b7c12468d450d59c4281f8f7fac6caf69ea65b5f4ef9ab8d3c1841bbb7f007e1129aafae48e3d217

[2022-01-28 15:28:46.359 WARNING  (unknown file)] [src/journal/block_validator.rs: 297] Block 84a0180d9800b6556deddd91e79636f18b901b14611441b2b7c12468d450d59c4281f8f7fac6caf69ea65b5f4ef9ab8d3c1841bbb7f007e1129aafae48e3d217 failed validation: Block 84a0180d9800b6556deddd91e79636f18b901b14611441b2b7c12468d450d59c4281f8f7fac6caf69ea65b5f4ef9ab8d3c1841bbb7f007e1129aafae48e3d217 failed validation: expected state hash 8a0150f15030450c4f7b1f5b86c9da9d987bbaf9861cdf576f22d9fb120e9967, validation found state hash b993bb9e6af7eee90cf7e6a39afc97efe1bf617028622e76b11b3a5b13929e9d

[2022-01-28 15:28:46.361 INFO     (unknown file)] [src/journal/block_scheduler.rs: 278] Predecessor 84a0180d9800b6556deddd91e79636f18b901b14611441b2b7c12468d450d59c4281f8f7fac6caf69ea65b5f4ef9ab8d3c1841bbb7f007e1129aafae48e3d217 marked invalid, marking descendant cf0aff85f29e2a80f7981fa6a3bae571b8270181c8fa3321ce0560fc4aab53c119f7c77b238403d9e8db1dbbfe97b25edf933ccaac2b6ba2757029e1071bf6cb invalid

^^^ repeats 35 times with different blocks

[2022-01-28 15:28:46.463 DEBUG    ffi] [src/journal/block_scheduler.rs: 226] Block 0fcaa984e7df703636e20bbe51d80adc06b1d459a2a65d8e73d1553d21149a5a46419e4d90be712ee62c095b842847cd2ffd3756d3d008e7f2bd8ee5e801a873 has invalid predecessor 84535b9fe204c0d140cf850e8ebeecd75a8d43f1c157a2f8353650c1f79aee33141c88bd2164dd1851d6b5024d1e6dcd1b16d697a8797db0a3f1558fa525154b, propagating invalidation

^^^ repeats 14723 times with different blocks 
```

The snapshot starts at block_num `1044071`, the node's tip is `1044073` and the rest of the network is at  `1058816`.

1044071+14723 = 1058794 < 1058816 which I think looks right.

What we see several hours later is that the same node hasn't synchronized at all with the rest of the network (and has mined just a few blocks on its own due to a high difficulty == 40).